src/grt/src/MakeWireParasitics.cpp

```diff
@@ -491,47 +491,31 @@ float MakeWireParasitics::getNetSlack(odb::dbNet* net)
 std::vector<int> MakeWireParasitics::routeLayerLengths(odb::dbNet* db_net) const
 {
   NetRouteMap& routes = grouter_->getRoutes();
-  std::vector<int> layer_lengths(grouter_->getMaxRoutingLayer() + 1);
+
+  // dbu wirelength for wires, via count for vias
+  std::vector<int> layer_lengths(tech_->getLayerCount());
+
   if (!db_net->getSigType().isSupply()) {
     GRoute& route = routes[db_net];
     std::set<RoutePt> route_pts;
     for (GSegment& segment : route) {
       if (segment.isVia()) {
-        route_pts.insert(
-            RoutePt(segment.init_x, segment.init_y, segment.init_layer));
-        route_pts.insert(
-            RoutePt(segment.final_x, segment.final_y, segment.final_layer));
+        auto& s = segment;
+        // Mimic makeRouteParasitics
+        int min_layer = min(s.init_layer, s.final_layer);
+        odb::dbTechLayer* cut_layer
+            = tech_->findRoutingLayer(min_layer)->getUpperLayer();
+        layer_lengths[cut_layer->getNumber()] += 1;
+        route_pts.insert(RoutePt(s.init_x, s.init_y, s.init_layer));
+        route_pts.insert(RoutePt(s.final_x, s.final_y, s.final_layer));
       } else {
         int layer = segment.init_layer;
-        layer_lengths[layer] += segment.length();
+        layer_lengths[tech_->findRoutingLayer(layer)->getNumber()]
+            += segment.length();
         route_pts.insert(RoutePt(segment.init_x, segment.init_y, layer));
         route_pts.insert(RoutePt(segment.final_x, segment.final_y, layer));
       }
     }
-    // Mimic MakeWireParasitics::makeParasiticsToPin functionality.
```

Comment:
Why is the following code no longer needed?

Reply:
My understanding is the grt guides reach the center of the instance not the individual pins, and this code is adding a small patch of route on top of it to meet the pin. In the process it's adding a redundant via (so we overcount), and I'm not sure the wirelength it adds is beneficial either.

If it's true that it doesn't help with the estimation (which can be confirmed empirically) then we should remove the same code from makeParasiticsPin to keep things in sync

Reply:
@povik A small clarification, but GRT guides don't reach the center of the instance. They reach the center of the GCell that encloses the pin shape. The distance from the center of the GCell to the pin location might be useful (more than 7 tracks, for example)

Reply:
What is the redundant via that is being added?

Reply:
Actually the redundant via exists only in `makeParasiticsToPin` (see `via_res` in the function). Here it wasn't being counted because we weren't counting vias at all, but now that we do, we would have a redundant via if we updated this piece of code instead of removing it.

Reply:
I'm afraid I'm not following.  My concern is that if estimate RC is creating a redundant via then it is incorrect and should be fixed.  Perhaps you and @eder-matheus can review this issue.